### ISSUE TITLE
Simple SDP fails to solve

### ISSUE THREAD

**Issue by Shuvomoy** [jump-dev/MosekTools.jl#80]
I recently updated some of my Julia packages related to optimization as follows:

```
[4076af6c] ↑ JuMP v0.21.10 ⇒ v0.22.1
[b8f27783] ↑ MathOptInterface v0.9.22 ⇒ v0.10.6
[1ec41992] ↑ MosekTools v0.9.4 ⇒ v0.10.0
[4076af6c] ↑ JuMP v0.21.10 ⇒ v0.22.1
[b8f27783] ↑ MathOptInterface v0.9.22 ⇒ v0.10.6
[1ec41992] ↑ MosekTools v0.9.4 ⇒ v0.10.0
```

Now after updating, I am getting error solving a very simple SDP, which I was able to solve without any problem prior to my update. My Mosek version is 9.3.10 (the latest). The SDP is as follows:

```julia
using JuMP, MosekTools, Mosek, LinearAlgebra

N = 3

model = Model(optimizer_with_attributes(Mosek.Optimizer))

# construct G ⪰ 0
@variable(model, G[1:N, 1:N], PSD)

@variable(model, F[1:N-1] >= 0)

A = [1 2 3; 4 5 6; 7 8 9]

b = [7;8]

@constraint(model,  F'*b + tr(G) <= 0)

@objective(model, Min, sum(F) - tr(G))

optimize!(model)

p_star = objective_value(model)

G_star = value.(G)

F_star = value.(F)

```

The error I am getting is as follows:
```

MethodError: no method matching set_column_name(::Mosek.Task, ::MosekTools.MatrixIndex, ::String)
Closest candidates are:
  set_column_name(::Mosek.Task, !Matched::MosekTools.ColumnIndex, ::String) at C:\Users\shuvo\.julia\packages\MosekTools\Ryr4X\src\variable.jl:44
  set_column_name(!Matched::MosekTools.Optimizer, !Matched::MathOptInterface.VariableIndex, ::String) at C:\Users\shuvo\.julia\packages\MosekTools\Ryr4X\src\variable.jl:47
in eval at base\boot.jl:360 
in top-level scope at junk_7.jl:22
in optimize! at JuMP\2IF9U\src\optimizer_interface.jl:167
in optimize! at JuMP\2IF9U\src\optimizer_interface.jl:167 
in var"#optimize!#121" at JuMP\2IF9U\src\optimizer_interface.jl:195
in optimize! at MathOptInterface\eoIu0\src\Utilities\cachingoptimizer.jl:285
in optimize! at MathOptInterface\eoIu0\src\MathOptInterface.jl:80 
in copy_to at MathOptInterface\eoIu0\src\Bridges\bridge_optimizer.jl:421 
in #copy_to#7 at MathOptInterface\eoIu0\src\Bridges\bridge_optimizer.jl:421 
in default_copy_to at MathOptInterface\eoIu0\src\Utilities\copy.jl:438
in pass_attributes at MathOptInterface\eoIu0\src\Utilities\copy.jl:69
in _pass_attribute at MathOptInterface\eoIu0\src\Utilities\copy.jl:84
in set at MathOptInterface\eoIu0\src\Bridges\bridge_optimizer.jl:1327 
in set at MosekTools\Ryr4X\src\variable.jl:296 
in set_column_name at MosekTools\Ryr4X\src\variable.jl:48
```

The code ran without any problem before updating.

**Comment by odow** [comment]
I don't have a Mosek installation so I can't work on this. But you could try adding this line to your example after `using ...`:
```julia
MosekTools.set_column_name(::Mosek.MSKtask, ::MosekTools.MatrixIndex, name::String) = name
```

If it works, it needs to be added here:
https://github.com/jump-dev/MosekTools.jl/blob/c176da6b3d801f27a0bdd671b5c8b1960e11c9cf/src/variable.jl#L44-L49

**Comment by Shuvomoy** [comment]
Yes, it works, I will create a pull request soon. Thank you @odow !

But I am very curious to know why the new version does not work as it is. I compared `variable.jl` file in `MosekTools 0.10.0` and `MosekTools 0.9.4`, and the only difference seems to be the name `MosekModel` in 0.9.4 has been replaced with `0.10.0`. Do you have any intuition behind why this might be happening?

**Comment by odow** [comment]
It's likely a change somewhere in JuMP or MOI that revealed this bug in MosekTools (i.e., we never tried to set the name of a matrix variable before?). We've had some massive changes between MOI 0.9 and 0.10, so it'd take a bit of work to track down in more detail.

**Comment by blegat** [comment]
Before, JuMP was not passing names to the inner optimizer (except in direct mode). With the new version of JuMP, it is passing the names.

**Comment by Shuvomoy** [comment]
Okay, that makes sense. I have created the pull request. I am curious, should not the function look similar to 

```
 function set_column_name(task::Mosek.MSKtask, mat::MatrixIndex, name::String) 
     putvarname(task, mat.column, name) 
 end 
```

analogous to the `ColumnIndex` case? I have no idea what is `set_column_name` doing, so I am curious.

**Comment by blegat** [comment]
In Mosek, scalar variables and matrix variables are separated. `putvarname(task, i, name)` set the names of the scalar variables of index `i`, also referred to as the `i`th column.
So that implementation of `set_column_name` is not correct. Last time I checked, Mosek does not support setting names for matrix variables.
@ulfworsoe Has this changed ?
Otherwise, the best we can do it is throw this error:
https://github.com/jump-dev/MathOptInterface.jl/blob/e1b14f818aa976a3057854a57c7cd86d694f58e1/src/attributes.jl#L79-L92

**Comment by blegat** [comment]
Another option is to return `false` here:
https://github.com/jump-dev/MosekTools.jl/blob/c176da6b3d801f27a0bdd671b5c8b1960e11c9cf/src/variable.jl#L292
and to throw
https://github.com/jump-dev/MathOptInterface.jl/blob/e1b14f818aa976a3057854a57c7cd86d694f58e1/src/attributes.jl#L63-L76
when a name is set to a matrix variable.

**Comment by Shuvomoy** [comment]
@blegat , I have tested MosekTools.jl with the following function: 

```
function set_column_name(task::Mosek.MSKtask, mat::MosekTools.MatrixIndex, name::String)
    putvarname(task, mat.matrix, name)
end
```
and this seems to be working. I have tested for three SDPs for my research, and it seems to be giving the correct result without throwing any error. Should I create a pull request with this (I closed my previous pr)? We can leave the issue open for now, until someone finds a proper fix.
